url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/ssamot/utilities/Configuration.java

```diff
@@ -0,0 +1,111 @@
+/*
+ *
+ * *** BEGIN LICENSE
+ *  Copyright (C) 2012 Spyridon Samothrakis spyridon.samothrakis@gmail.com
+ *  This program is free software: you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 3, as published
+ *  by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful, but
+ *  WITHOUT ANY WARRANTY; without even the implied warranties of
+ *  MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * *** END LICENSE
+ *
+ */
+
+package ssamot.utilities;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.MissingResourceException;
+import java.util.Properties;
+
+public class Configuration {
+	private static Configuration theInstance;
+	private static String PATH = "./";
+	private static String PROPERTIES_FILENAME = "main.config";
+	private Properties props = null;
+	
+
+
+	public String getRuntime_directory() {
+		return getPropertyAsString("runtime.directory");
+	}
+
+	public String getCma_configuration_file() {
+		return getRuntime_directory() + getPropertyAsString("cma.configuration.file") ;
+	}
+
+	public Boolean istCocoBbopNoisy() {
+		return getPropertyAsBoolean("coco.bbop.noisy") ;
+	}
+	
+	
+	private Configuration() throws IOException {
+
+		FileInputStream propFile = new FileInputStream(PATH
+				+ PROPERTIES_FILENAME);
+		props = new Properties(System.getProperties());
+		props.load(propFile);
+
+	}
+
+	public static Configuration getConfiguration() {
+		synchronized (Configuration.class) {
+			if (theInstance == null) {
+				try {
+					theInstance = new Configuration();
+				} catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)